### PR TITLE
rename a bunch of things from slash- to interaction-

### PR DESCRIPTION
**Pull request by akiraveliara**
as discussed in https://canary.discord.com/channels/379378609942560770/1202000142170062928/1204476616051269682

### CODE REVIEW

File: DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionAutoCompleteProviderAttribute.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace DSharpPlus.Commands.Processors.SlashCommands.Attributes;
 using Microsoft.Extensions.DependencyInjection;
 
 [AttributeUsage(AttributeTargets.Parameter, Inherited = false, AllowMultiple = false)]
-public class SlashAutoCompleteProviderAttribute(Type autoCompleteType) : Attribute
+public class InteractionAutoCompleteProviderAttribute(Type autoCompleteType) : Attribute
```
Comment (OoLunar) [inline DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionAutoCompleteProviderAttribute.cs:10]: Autocomplete is currently only available for slash commands. Should remain as `SlashAutoCompleteProviderAttribute` imo.

File: DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionChannelTypesAttribute.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@ namespace DSharpPlus.Commands.Processors.SlashCommands.Attributes;
 using System;
 
 [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
-public sealed class SlashChannelTypesAttribute(params ChannelType[] channelTypes) : Attribute
+public sealed class InteractionChannelTypesAttribute(params ChannelType[] channelTypes) : Attribute
```
Comment (OoLunar) [inline DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionChannelTypesAttribute.cs:6]: As above

File: DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionChoiceProviderAttribute.cs [hunk 3]
```diff
@@ -8,7 +8,7 @@ namespace DSharpPlus.Commands.Processors.SlashCommands.Attributes;
 using Microsoft.Extensions.DependencyInjection;
 
 [AttributeUsage(AttributeTargets.Parameter, Inherited = false, AllowMultiple = false)]
-public class SlashChoiceProviderAttribute(Type providerType) : Attribute
+public class InteractionChoiceProviderAttribute(Type providerType) : Attribute
```
Comment (OoLunar) [inline DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionChoiceProviderAttribute.cs:11]: As above

File: DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionMinMaxAttribute.cs
Comment (OoLunar) [inline DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionMinMaxAttribute.cs:1]: As above

File: DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionMinMaxLengthAttribute.cs [hunk 4]
```diff
@@ -6,7 +6,7 @@ namespace DSharpPlus.Commands.Processors.SlashCommands.Attributes;
 /// Determines the minimum and maximum length that a parameter can accept.
 /// </summary>
 [AttributeUsage(AttributeTargets.Parameter)]
-public sealed class SlashMinMaxLengthAttribute : Attribute
+public sealed class InteractionMinMaxLengthAttribute : Attribute
```
Comment (OoLunar) [inline DSharpPlus.Commands/Processors/SlashCommands/Attributes/InteractionMinMaxLengthAttribute.cs:9]: As above

**Review by OoLunar — CHANGES_REQUESTED** [review]
Food for thought

**Review by Plerx2493 — APPROVED** [review]
LGTM, if lunars requests getting applied

**Review by OoLunar — APPROVED** [review]

**Pull request merged**